### ISSUE TITLE
enum_variant_names false positive when first word has common prefix as part of word

### ISSUE THREAD

**Issue by tspiteri** [rust-lang/rust-clippy#8090]
### Summary

The lint warns if all variants have the same prefix word. If the first variant has the prefix word at the start of its first word, it wrongly triggers the lint even if the first word is longer than the prefix word. If this variant is not the first variant, the warning is not triggered.

### Lint Name

enum_variant_names

### Reproducer

I tried this code:

```rust
enum _E {
    Normal,
    NoLeft,
    NoRight,
}
```

I saw this happen:

```
warning: all variants have the same prefix: `No`
 --> src/lib.rs:1:1
  |
1 | / enum _E {
2 | |     Normal,
3 | |     NoLeft,
4 | |     NoRight,
5 | | }
  | |_^
  |
  = note: `#[warn(clippy::enum_variant_names)]` on by default
  = help: remove the prefixes and use full paths to the variants instead of glob imports
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#enum_variant_names
```

I expected no warning.

### Version

```text
rustc 1.58.0-beta.2 (0e07bcb68 2021-12-04)
binary: rustc
commit-hash: 0e07bcb68b82b54c0c4ec6fe076e9d75b02109cf
commit-date: 2021-12-04
host: x86_64-unknown-linux-gnu
release: 1.58.0-beta.2
LLVM version: 13.0.0

clippy 0.1.58 (0e07bcb 2021-12-04)
```

**Comment by dswij** [comment]
Huh 🤔 I can take a look on this

@rustbot claim
